core/iwasm/compilation/aot_emit_memory.c

```diff
@@ -202,9 +203,32 @@ aot_check_memory_overflow(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx,
                     goto fail;
                 }
             }
+            if (alignp != NULL) {
+                /*
+                 * A note about max_align below:
+                 * the assumption here is the base address of a linear memory
+                 * has the natural alignment. for platforms using mmap, it can
+                 * be even larger. for now, use a conservative value.
+                 */
+                const int max_align = 4;
+                int shift = ffs((int)(unsigned int)mem_offset);
+                if (shift == 0) {
+                    *alignp = max_align;
+                }
+                else {
+                    unsigned int align = 1 << (shift - 1);
+                    if (align > max_align) {
+                        align = max_align;
+                    }
+                    *alignp = align;
+                }
+            }
             return maddr;
         }
     }
+    if (alignp != NULL) {
```

Comment:
Can change to `else if` if we handle all the situations for `if (LLVMIsEfficientConstInt(addr))`, or how about putting it before L175?

Reply:
done